app/src/main/java/org/worldbank/transport/driver/utilities/DriverUtilities.java

```diff
@@ -217,9 +221,36 @@ public static boolean localeIsRTL() {
     }
 
     public static boolean localeIsRTL(Locale locale) {
-        Log.d(LOG_LABEL, locale.getDisplayName());
+        Log.d(LOG_LABEL, "Locale " + locale.getDisplayName() + " language " + Locale.getDefault().getLanguage());
+
         int direction = Character.getDirectionality(locale.getDisplayName().charAt(0));
         return  direction == Character.DIRECTIONALITY_RIGHT_TO_LEFT_ARABIC ||
                 direction == Character.DIRECTIONALITY_RIGHT_TO_LEFT;
     }
+
+    /**
+     * Check if the system language is Arabic. Use to decide whether to use Hijri date display.
+     *
+     * @return True if device language is Arabic.
+     */
+    public static boolean languageIsArabic() {
+        return Locale.getDefault().getLanguage().startsWith("ar");
+    }
+
+    /**
+     * Convert a date to its Hijri calendar representation.
+     * Uses Joda for the conversion, as it is already included as a dependency.
+     *
+     * @param date ISO date/time to format
+     * @return Hijri date string
+     */
+    public static String formatDateAsHijri(Date date) {
+        final Chronology hijri = IslamicChronology.getInstanceUTC();
```

Comment:
Unfortunately this is probably not going to work; Saudi Arabia uses a different flavor of Islamic calendar (called Umm al-Qura) which is apparently not available in Joda. We encountered this in the Python and JS code as well -- the Islamic calendar is consistently off by a day when compared to the Umm al-Qura calendar. I'm not sure exactly what our options are technologically, but this SO answer seems to be on point, and it looks like there are a few potential solutions: http://stackoverflow.com/a/21030128